src/transform/mod.rs

```diff
@@ -43,10 +40,12 @@ pub fn transform_no_free_vars(
             if let Some(wildcards) = wildcards {
                 rt = substitute_wildcards::transform_one(rt, wildcards);
             }
-            // before anything, we have to desugar the syntax
+            // We desugar destructuring before other transformations, as this step generates new
+            // record contracts and terms that must be themselves transformed.
             let rt = desugar_destructuring::transform_one(rt);
-            // We need to do contract generation before wrapping stuff in variables
-            let rt = apply_contracts::transform_one(rt)?;
+            // We need to do contract generation before the share_normal_form transformation,
+            // because gen_pending_contracts generates record contracts
```

Comment:
```suggestion
            // We need to do contract generation before the share normal form transformation,
            // because `gen_pending_contracts` generates record contracts
```

Reply:
In the end share normal form is better spelled out than quoted, I think.